projects/plutus/inc/current_sense.h

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+// Current sense module for the LTC2484 ADC.
+// Requires gpio, interrupts, and soft timers to be initialized
+
+#include "ltc_adc.h"
+#include "status.h"
+
+// If the adc has not produced valid data (i.e. timeout), then |current| will be a null pointer
+typedef void (*CurrentSenseCallback)(int32_t current, void *context);
+
+typedef struct {
+  int32_t voltage;
+  int32_t current;
+} CurrentSenseValue;
+
+// Line data for two-point calibration
+typedef struct {
+  CurrentSenseValue zero_point;
+  CurrentSenseValue max_point;
+} CurrentSenseCalibrationData;
+
+typedef struct {
+  LtcAdcStorage adc_storage;
+  CurrentSenseCalibrationData data;
+  CurrentSenseValue value;
+  bool data_valid;
+  int32_t offset;
+  bool offset_pending;
+  CurrentSenseCallback callback;
+  CurrentSenseCallback fault_callback;  // Called in the event of an adc timeout
```

Comment:
No

make a new callback

Reply:
You should clarify that you mean definition. Fixed